resources/static/js/account.js

```diff
@@ -86,72 +94,90 @@ function showRamp() {
 }
 
 async function checkWeb3Network() {
-  let required_chain_id = CHAIN_ID;
-  await connectWeb3();
-  web3.version.getNetwork(function (error, chain_id) {
-    if (error) {
-      console.error(error);
-    }
-
-    if (chain_id != required_chain_id) {
-      let current_chain_name = CHAIN_ID_MAPPING[chain_id];
-      let required_chain_name = CHAIN_ID_MAPPING[required_chain_id];
-      alert(
-        `Web3 Browser connected to ${current_chain_name}, please change to ${required_chain_name}.`
-      );
-    }
-  });
+  const requiredChainID = CHAIN_ID;
+  const currentChainID = parseInt(await provider.request({ method: 'eth_chainId' }), 16);
+  if (currentChainID != requiredChainID) {
+    const currentChainName = CHAIN_ID_MAPPING[currentChainID];
+    const requiredChainName = CHAIN_ID_MAPPING[requiredChainID];
+    alert(
+      `Web3 Browser connected to ${currentChainName}, please change to ${requiredChainName}.`
+    );
+    return false;
+  }
+  return true;
 }
 
-function makeWeb3Transaction(w3, transaction_data) {
-  w3.eth.sendTransaction(transaction_data, function (error, result) {
-    if (result) {
-      // result is the transaction hash
-      trackTransaction(result, CONFIGURATION_FILE_NAME);
+async function connectWeb3() {
+  let accounts = [];
+  try {
+    accounts = await provider.request({ method: 'eth_accounts' });
+    if (accounts.length === 0) {
+      accounts = await provider.request({ method: 'eth_requestAccounts' });
     }
-
-    if (error) {
+  } catch (error) {
+    if (error.code === 4001) {
+      // EIP-1193 userRejectedRequest error
+      alert('Permissions to the web3 provider\'s accounts needed in order to continue.');
+    } else {
+      alert('Not able to connect to the web3 provider.');
       console.error(error);
     }
-  });
+  }
+  return accounts;
+}
+
+async function makeWeb3Transaction(transactionParams) {
+  let transactionHash;
+  try {
+    transactionHash = await provider.request({ method: 'eth_sendTransaction', params: [transactionParams]});
+  } catch (error) {
+    console.error(error);
+  }
+  if (transactionHash) {
+    trackTransaction(transactionHash, CONFIGURATION_FILE_NAME);
+  }
 }
 
 async function sendEthViaWeb3() {
-  await checkWeb3Network();
-  let web3 = window.web3;
-  let sender_address =
-    (window.ethereum && window.ethereum.selectedAddress) ||
-    web3.eth.defaultAccount;
-  let gasPrice;
+  const correctNetwork = await checkWeb3Network();
+  if (!correctNetwork) {
+    return;
+  }
 
+  const accounts = await connectWeb3();
+  if (accounts.length === 0) {
+    return;
```

Comment:
Can this also be the case when the `eth_requestAccounts` was successful but the list is still empty?
Would this scenario need communication with the user?

Reply:
From the EIP 1102:
> Browsers MUST include at least one account if the eth_requestAccounts promise is resolved.

The list will only be empty if there was an error and in this case we show an alert.